src/Compilers/Core/CodeAnalysisTest/MetadataReferences/MetadataReferenceTests.cs

```diff
@@ -56,18 +56,38 @@ public void CreateFrom_Errors()
         }
 #endif
 
-        [Fact]
-        public void CreateFromImage()
+        [Theory, CombinatorialData]
+        public void CreateFromImage_Assembly(bool immutableArray)
         {
-            var r = MetadataReference.CreateFromImage(ResourcesNet451.mscorlib);
+            var peImage = ResourcesNet451.mscorlib;
+            var r = immutableArray
+                ? MetadataReference.CreateFromImage(peImage.AsImmutable())
+                : MetadataReference.CreateFromImage(peImage.AsEnumerable());
 
+            Assert.IsAssignableFrom<AssemblyMetadata>(r.GetMetadata());
             Assert.Null(r.FilePath);
             Assert.Equal(CodeAnalysisResources.InMemoryAssembly, r.Display);
             Assert.Equal(MetadataImageKind.Assembly, r.Properties.Kind);
             Assert.False(r.Properties.EmbedInteropTypes);
             Assert.True(r.Properties.Aliases.IsEmpty);
         }
 
+        [Theory, CombinatorialData]
+        public void CreateFromImage_Module(bool immutableArray)
+        {
+            var peImage = TestResources.MetadataTests.NetModule01.ModuleCS00;
```

Comment:
>TestResources.MetadataTests.NetModule01.ModuleCS00 [](http://example.com/codeflow?start=26&length=50)

If we expect the operation to succeed with an assembly image, let's cover this case as well. For all API flavors.